### PR TITLE
 Remove client.ListOptions.Raw hack 

### PR DESCRIPTION
**Pull request by vincepri**
**What this PR does / why we need it**:
Removes a temporary hack necessary to run tests with controller-runtime Fake client.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by vincepri** [comment]
Stumbled upon this while working on #41

**Comment by detiber** [comment]
Nice find! lgtm, but holding off on triggering bot until @roberthbailey can review

File: config/crds/cluster_v1alpha1_machine.yaml [hunk 1]
```diff
@@ -42,6 +42,22 @@ spec:
                 to indicate what labels, annotations, name prefix, etc., should be
                 used when creating the Node.
               type: object
+            providerID:
```
Comment (roberthbailey) [inline config/crds/cluster_v1alpha1_machine.yaml]: Do these changes get clobbered if we change the apis and re-run make?

Comment (vincepri) [reply config/crds/cluster_v1alpha1_machine.yaml]: Yeah that's correct. Ideally we should check this in CI like we do for the clientset.

**Pull request merged**